src/lib/slp/dialogs/service_selection.rb

```diff
@@ -0,0 +1,163 @@
+      # @param heading         [String] Dialog's heading
+      # @param description     [String] Dialog's description (to be shown on top of the list)
+      # @param no_selected_msg [String] Message to be shown when no service was selected
+      # @param initial         [SlpServiceClass::Service] initially selected service
+      def initialize(services: [], heading: nil, description: nil, no_selected_msg: nil, initial: nil)
+        super()
+
+        textdomain "registration"
+
+        @services = services
+        @heading = heading || _("Service selection")
+        @description = description || _("Select a detected service from the list.")
+        @no_selected_msg = no_selected_msg || _("No service was selected.")
+        @initial = initial || services.first
+      end
+
+      # Handler for the Ok button
+      #
+      # If no option was selected, a error message is shown.
+      def ok_handler
+        selected = Yast::UI.QueryWidget(Id(:services), :CurrentButton)
+        if !selected
+          Yast::Report.Error(no_selected_msg)
+        else
+          finish_dialog(services[selected.to_i])
+        end
+      end
+
+      # Handler for the cancel button
+      def cancel_handler
+        finish_dialog(:cancel)
+      end
+
+    protected
+
+      # Dialog's initial content
+      #
+      # @return [Yast::Term] Content
+      def dialog_content
+        MarginBox(2, 0.5,
+          VBox(
+            # popup heading (in bold)
+            Heading(heading),
+            VSpacing(0.5),
+            Label(description),
+            VSpacing(0.5),
+            RadioButtonGroup(
+              Id(:services),
+              Left(
+                HVSquash(
+                  VBox(*services_radio_buttons)
+                )
+              )
+            ),
+            VSpacing(Opt(:vstretch), 1),
+            button_box
+          ))
+      end
+
+      # Dialog options
+      #
+      # @return [Yast::Term] Dialog's options
+      def dialog_options
+        Yast::Term.new(:opt, :decorated)
+      end
+
+      # Return dialog's buttons
+      #
+      # @return [Yast::Term] Buttons' description
+      def button_box
+        ButtonBox(
+          PushButton(Id(:ok), Opt(:default), Yast::Label.OKButton),
+          PushButton(Id(:cancel), Yast::Label.CancelButton)
+        )
+      end
+
+      # Return service radio buttons
+      #
+      # @return [Yast::Term] Service radio button's description
+      def services_radio_buttons
+        services.map.with_index do |service, idx|
+          Left(
+            RadioButton(Id(idx.to_s), service_to_description(service), service == initial)
+          )
+        end
+      end
+
+      # Return the service description to be shown to the user
+      #
+      # @return [String] Service description
+      def service_to_description(service)
+        url = service.slp_url.sub(service.slp_type, service.protocol)
```

Comment:
I considered also that option, but I was unsure about how useful it could be. So I decided to go from the `registration` to the `slp` case (and not to a generic case).